README.md

```diff
@@ -1,54 +1,81 @@
 # simplesamlphp-module-assurance
 A SimpleSAMLphp module for determining and indicating the Level of Assurance (LoA) of an authentication event.
-Specifically the module determines the LoA baced on:
-* `candidates`, if the represented LoA attribute of the authenticating user contains any value of the allowed values, then the module will skip execution.
-* `idpTags`, Identity Providers (IdPs) that are providing a higher level of assuranse for the authincating user can be marked, by adding the attribute `'tags' => array (0 => 'exampleTag',)` in the IdP's metadata in `saml20-idp-remote.php`.
-* `idpPolicies`, if the 'eduPersonAssurance' attribute of the authenticating user contains any value of this list, then the user will grand higher LoA.
-* `entitlements`, if the 'eduPersonEntitlement' attribute of the authenticating user contains any value of this list, then the user will grant higher LoA.
 
 To this end, the LoA that the module will determine for the authnticating user, will be saved in the SAML 2 response.
 
 ## Module configuration
 The following classes are needed to configure, in order to execute the module.
 ### DynamicAssurance
+This filter evaluates the assurance values based on specific attribute mapping rules. The evaluation process can optionally take into account the tag(s) of the authenticating Identity Provider.
 #### Configuration
 The following authproc filter configuration options are supported:
-  * `entitlementWhitelist`: Required, an array of strings containing values that can get higher assurance.
-  * `attribute`: Optional, a string that defines the name of the attribute that will store the value of LoA. Defaults to `eduPersonAssurance`.
-  * `candidates`: Optional, an array of strings that contains the allowed LoA values.
-  * `defaultAssurance`: Optional, a string to use as the default value of LoA. Defaults to `https://www.example.org/low`.
-  * `defaultElevatedAssurance`: Optional, a string to use as the default value of LoA. Defaults to `https://www.example.org/Substantial`.
-  * `idpPolicies`: Optional, an array of stings that contains the allowed IdP Policies.
-  * `idpTags`: Optional, an array of stings that contains the allowed IdP Tags.
+  * `attribute`: _Optional_, a string that defines the name of the attribute that will store the value of LoA. Defaults to `eduPersonAssurance`.
+  * `attributeMap`: _Optional_,  a map whose `keys` identify attribute names whose values can be mapped to assurance values. Represented LoA attributes, of the authenticating user, contain any of the list of the allowed values, then the module will append these values into the State. Defaults to
```

Comment:
* `attributeMap`: _Optional_, a map whose `keys` identify attribute names whose values can be mapped to assurance values. For each identified attribute, you can specify an array of string values to match. For each matching value, the filter appends the specified assurance values to the assurance attribute. Defaults to

Reply:
Done